src/m365/pp/commands/dataverse/dataverse-table-list.spec.ts

```diff
@@ -0,0 +1,288 @@
+          return Promise.resolve(envResponse);
+        }
+      }
+
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.0/EntityDefinitions?%24select=MetadataId%2CIsCustomEntity%2CIsManaged%2CSchemaName%2CIconVectorName%2CLogicalName%2CEntitySetName%2CIsActivity%2CDataProviderId%2CIsRenameable%2CIsCustomizable%2CCanCreateForms%2CCanCreateViews%2CCanCreateCharts%2CCanCreateAttributes%2CCanChangeTrackingBeEnabled%2CCanModifyAdditionalSettings%2CCanChangeHierarchicalRelationship%2CCanEnableSyncToExternalSearchIndex&%24filter=(IsIntersect%20eq%20false%20and%20IsLogicalEntity%20eq%20false%20and%0APrimaryNameAttribute%20ne%20null%20and%20PrimaryNameAttribute%20ne%20%27%27%20and%20ObjectTypeCode%20gt%200%20and%0AObjectTypeCode%20ne%204712%20and%20ObjectTypeCode%20ne%204724%20and%20ObjectTypeCode%20ne%209933%20and%20ObjectTypeCode%20ne%209934%20and%0AObjectTypeCode%20ne%209935%20and%20ObjectTypeCode%20ne%209947%20and%20ObjectTypeCode%20ne%209945%20and%20ObjectTypeCode%20ne%209944%20and%0AObjectTypeCode%20ne%209942%20and%20ObjectTypeCode%20ne%209951%20and%20ObjectTypeCode%20ne%202016%20and%20ObjectTypeCode%20ne%209949%20and%0AObjectTypeCode%20ne%209866%20and%20ObjectTypeCode%20ne%209867%20and%20ObjectTypeCode%20ne%209868)%20and%20(IsCustomizable%2FValue%20eq%20true%20or%20IsCustomEntity%20eq%20true%20or%20IsManaged%20eq%20false%20or%20IsMappable%2FValue%20eq%20true%20or%20IsRenameable%2FValue%20eq%20true)&api-version=9.1`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return Promise.resolve(dataverseResponse);
+        }
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    await command.action(logger, { options: { debug: true, environment: '4be50206-9576-4237-8b17-38d8aadfaa36', asAdmin: true } });
+    assert(loggerLogSpy.calledWith(dataverseResponse.value));
+  });
+
+  it('correctly handles API OData error', async () => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if ((opts.url === `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/4be50206-9576-4237-8b17-38d8aadfaa36?api-version=2020-10-01&$select=properties.linkedEnvironmentMetadata.instanceApiUrl`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return Promise.resolve(envResponse);
+        }
+      }
+
+      return Promise.reject({
+        error: {
+          'odata.error': {
+            code: '-1, InvalidOperationException',
+            message: {
+              value: `Resource '' does not exist or one of its queried reference-property objects are not present`
+            }
+          }
+        }
+      });
+    });
+
+    try {
+      await command.action(logger, { options: { debug: false, environment: '4be50206-9576-4237-8b17-38d8aadfaa36' } });
```

Comment:
I'm missing an assert.fail here.

Reply:
Not sure if I understand correctly; implemented the `assert.rejects` but it still ends up in the catch

Reply:
I mean what you do in the planner utility spec file: If (for whatever reason) the code does not end up in the catch statement, it should assert that the test was not successful. As it should fail....

Reply:
That totally makes more sense, implemented right away!